src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateGetHashCodeMethod.cs

```diff
@@ -131,7 +131,7 @@ internal static partial class SyntaxGeneratorExtensions
 
             const string HashCodeName = "hashCode";
             statements.Add(!useInt64
-                ? factory.SimpleLocalDeclarationStatement(compilation.GetSpecialType(SpecialType.System_Int32), HashCodeName, CreateLiteralExpression(factory, initHash))
+                ? factory.SimpleLocalDeclarationStatement(compilation.GetSpecialType(SpecialType.System_Int32)!, HashCodeName, CreateLiteralExpression(factory, initHash))
```

Comment:
Looking through our code it seems there will be lots of places we need the `!` operator for `GetSpecialType`. That is unfortunate because it is very difficult to search for places where we use `GetSpecialType` in a "unsafe" way. One option I considered is adding a helper here like `GetSpecialTypeOrThrow`. That is much more declarative about the intent as well as being very searchable. Thoughts?

Reply:
`GetRequiredSpecialType` would be the best match for the IDE code in similar places.

Reply:
Did a quick check with compiler team and the `OrThrow` suffix was preferred. 

---
In reply to: [359948335](https://github.com/dotnet/roslyn/pull/40493#discussion_r359948335) [](ancestors = 359948335)